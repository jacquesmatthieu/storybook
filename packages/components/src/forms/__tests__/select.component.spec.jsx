import React from 'react';
import { shallow } from 'enzyme';

import Select from '../select.component';

describe('Select', () => {
  let props;

  beforeEach(() => {
    props = {
      input: {
        name: 'foo',
      },
      meta: {
        valid: true,
      },
      options: [
        ['foo', 'Foo'],
        ['bar', 'Bar'],
      ],
    };
  });

  const getWrapper = () => shallow(<Select {...props} />);

  it('should render with default props', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with specified props', () => {
    // Given
    props.label = 'bar';
    props.meta = {
      error: true,
      touched: true,
    };

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
