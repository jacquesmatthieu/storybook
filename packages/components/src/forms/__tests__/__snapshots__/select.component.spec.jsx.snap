// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render with default props 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label=""
  name="foo"
>
  <selectcomponent__StyledSelect
    error={false}
    id="foo"
    name="foo"
    valid={true}
  >
    <option
      key="foo"
      value="foo"
    >
      Foo
    </option>
    <option
      key="bar"
      value="bar"
    >
      Bar
    </option>
  </selectcomponent__StyledSelect>
  <selectcomponent__SelectIcon
    name="arrow"
    size={15}
    stroke="grey.dark"
    strokeWidth={1}
  />
</InputGroup>
`;

exports[`Select should render with specified props 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label="bar"
  name="foo"
>
  <selectcomponent__StyledSelect
    error={true}
    id="foo"
    name="foo"
  >
    <option
      key="foo"
      value="foo"
    >
      Foo
    </option>
    <option
      key="bar"
      value="bar"
    >
      Bar
    </option>
  </selectcomponent__StyledSelect>
  <selectcomponent__SelectIcon
    name="arrow"
    size={15}
    stroke="grey.dark"
    strokeWidth={1}
  />
</InputGroup>
`;
